BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
+			
+        var data = client.GetStringAsync(MainCurrenciesUrl).Result;
```

Comment:
* Here we are returning the response body as a string, but in the case of an error 404, it won't return a body and we'll likely get an error.
We need to deal with these possibilities and, especially regarding to error handling. Consider checking: https://api.cnb.cz/cnbapi/swagger-ui.html#/%2Fexrates/dailyUsingGET_1 
* We could use the verb **GetAsync()**, which will return an **HttpResponseMessage** and, in case of unsuccessful status code, we can return a specific error.

Reply:
If it's an unsuccessful status code, we could read the response and translate it for our client. An example could be:

```
public struct CnbRatesFailedResponse
{
    [JsonPropertyName("description")]
    public string ErrorDescription { get; set; }

    [JsonPropertyName("errorCode")]
    public string ErrorCode { get; set; }
}
```